playbooks/infra/reporting/roles/junit2json/tasks/convert.yml

```diff
@@ -0,0 +1,46 @@
+# SPDX-License-Identifier: Apache-2.0
+---
+# tasks file for redhatci.ocp.junit2json
+# task list: convert.yml converts xml into JSON
+
+- name: Read file content
+  ansible.builtin.set_fact:
+    junit2json_xml_report_content: "{{ lookup('ansible.builtin.file', file_name) }}"
+
+- name: Update junit2_result_data junit2_do_merge=true
+  ansible.builtin.set_fact:
+    junit2_result_data: "{{ junit2json_xml_report_content }}"
+
+- name: Setup JSON report file name
+  ansible.builtin.set_fact:
+    junit2_json_report_path: "{{ file_name | basename | regex_replace('\\.xml$', '.json') if file_name.endswith('.xml') else file_name | basename + '.json' }}"
+
+- name: Set junit2_output_report_path
+  ansible.builtin.set_fact:
+    junit2_output_report_path: "{{ junit2_output_dir }}/{{ junit2_json_report_path }}"
+
+- name: Update output variable junit2_json_reports_list
+  ansible.builtin.set_fact:
+    junit2_json_reports_list: "{{ junit2_json_reports_list + [junit2_output_report_path] }}"
+
+- name: Ensure junit2_output_dir is created
+  ansible.builtin.include_tasks:
+    file: ensure-dir.yml
+  loop:
+    - "{{ junit2_output_dir }}"
+  loop_control:
+    loop_var: folder_path
```

Comment:
why is this a loop ? are we creating more than one folder? 

also if you only create the folder once no need to put it in a new file / playboolk

Reply:
@eifrach 👍  thank you, good catch, it is a left-over